File: bot-api/src/main/java/gg/mineral/bot/api/math/trajectory/throwable/EnderPearlTrajectory.java
package gg.mineral.bot.api.math.trajectory.throwable;

import gg.mineral.bot.api.world.ClientWorld;

public class EnderPearlTrajectory extends ThrowableEntityTrajectory {

    public EnderPearlTrajectory(ClientWorld world, double x, double y, double z, float yaw, float pitch, CollisionFunction collisionFunction) {
        super(world, x, y, z, yaw, pitch, collisionFunction);
    }

    @Override
    public float offset() {
        return 0.0F;
    }

    @Override
    public float power() {
        return 1.5F;
    }

    @Override
    public float gravity() {
        return 0.03F;
    }
}

File: bot-api/src/main/kotlin/gg/mineral/bot/api/math/trajectory/throwable/EnderPearlTrajectory.kt
package gg.mineral.bot.api.math.trajectory.throwable

import gg.mineral.bot.api.math.trajectory.Trajectory
import gg.mineral.bot.api.world.ClientWorld

class EnderPearlTrajectory(
    world: ClientWorld,
    x: Double,
    y: Double,
    z: Double,
    yaw: Float,
    pitch: Float,
    collisionFunction: Trajectory.CollisionFunction
) :
    ThrowableEntityTrajectory(world, x, y, z, yaw, pitch, collisionFunction) {
    override fun offset(): Float {
        return 0.0f
    }

    override fun power(): Float {
        return 1.5f
    }

    override fun gravity(): Float {
        return 0.03f
    }
}
